### ISSUE TITLE
mmseqs_update: Problem with data file. 

### ISSUE THREAD

**Issue by narsapuramvijaykumar** [soedinglab/mmseqs2#4]
When I am trying to run mmseq_update on some data set below error was popping up.
ERROR:
mmseqs_update: Problem with data file. Is it empty or is another process readning it?: Invalid argument
commons/DBReader.cpp:49 ffindex_index_parse: /anno/narsapvi/fern_clustering/run/tmp/A.index: Invalid argument
What can be the possible solution for this..!!

**Comment by milot-mirdita** [comment]
Hi and thank you for your bug report.

Would you mind switching to MMseqs2, our new and much improved release? MMseqs1 is not supported anymore.

With that said, it looks like your A.index file is corrupted. Check if it has the expected number of lines (=number of clusters) and also if for every line there are three values.

**Comment by narsapuramvijaykumar** [comment]
I am upgrading my system(cmake, g++ etc) to make suitable for MMseq2 installation. Once it is done I will switch to MMseqs2.

The A.index file is totally empty. I am unable to figure out the possible solution.
I have tried multiple times with all input files formatted correctly.

Thanks  in advance.

**Comment by martin-steinegger** [comment]
You can use MMseqs2 without any requirements using the static compiled verison. 
```
    wget https://mmseqs.com/latest/mmseqs-static_sse41.tar.gz 
    tar xvzf mmseqs-static_avx2.tar.gz
    export PATH=$(pwd)/mmseqs/bin/:$PATH
```

Can you send me how you called mmseqs_update? What kind of data did you use?

**Comment by narsapuramvijaykumar** [comment]
Firstly  I have converted the fasta format to ffindex 
fastatoffindex old_DB.fasta old_DB
fastatoffindex new_DB.fasta new_DB
mmseq_cluster old_DB old_DB_clu tmp/ --cascaded
Above generated  old_DB_clu, I have given as oldDB_clustering to mmseq_update
mmseq_update old_DB new_DB old_DB_clu new_DB_clu tmp/
The above last command throws below error.
mmseqs_update: Problem with data file. Is it empty or is another process readning it?: Invalid argument
commons/DBReader.cpp:49 ffindex_index_parse: tmp//A.index: Invalid argument

**Comment by martin-steinegger** [comment]
Is it possible that old_DB and new_DB do not have any keys in common? A.index should be just empty if there is no overlap between both databases.

**Comment by narsapuramvijaykumar** [comment]
There is no identical fasta ids/keys between old_DB and new_DB

**Comment by martin-steinegger** [comment]
Okey we do not handle this. We will have a look if this also occures in MMseqs2.  
You could add one sequences from the old_DB to the new DB as a quick fix.

**Comment by narsapuramvijaykumar** [comment]
I have tried to update my clusters with the static compiled verison of mmseq2.
command used for updating clusters.
```
mmseqs clusterupdate old_DB new_DB old_DB_clu new_DB_clu tmp2/
```
Errors noticed at various levels:
```
Failed to mmap memory dataSize=0 File=tmp2//NEWDB.newSeqs
Failed to mmap memory dataSize=0 File=tmp2//NEWDB.newSeqs
mv: cannot stat `/tmp2/aln_4': No such file or directory
Failed to mmap memory dataSize=0 File=tmp2//NEWDB.newSeqs
Could not open data file tmp2//newSeqsHits.swapped.all!
awk: cmd. line:1: fatal: cannot open file `tmp2//newSeqsHits.index' for reading (No such file or directory)
mv: cannot stat `tmp2//aln_*': No such file or directory
mv: cannot stat `tmp2//pref_*': No such file or directory
mv: cannot stat `tmp2//clu_*': No such file or directory
mv: cannot stat `tmp2//input_*': No such file or directory
Failed to mmap memory dataSize=0 File=tmp2//toBeClusteredSeparately
Failed to mmap memory dataSize=0 File=tmp2//toBeClusteredSeparately
awk: cmd. line:1: fatal: cannot open file `tmp2//clu_redundancy.index' for reading (No such file or directory)
Failed to mmap memory dataSize=0 File=tmp2//toBeClusteredSeparately
Could not open data file tmp2//input_step_redundancy!
Could not open data file tmp2//input_step_redundancy!
Could not open data file tmp2//input_step_redundancy!
Failed to mmap memory dataSize=0 File=tmp2//toBeClusteredSeparately
mv: cannot stat `tmp2//clu': No such file or directory
mv: cannot stat `tmp2//clu.index': No such file or directory
```

**Comment by narsapuramvijaykumar** [comment]
After adding a single sequence record from old_DB to new_DB, the mmseq program seems to be working.
Thanks for the inputs martin-steinegger.
In future, if the above case (no identical keys between old and new DB) is handled either in MMseq or MMseqs2 that will be a great help for us.

**Comment by martin-steinegger** [comment]
We fixed the updating in MMseqs2. In order to use MMseqs2 you need to recluster you database since the database format changed. To recluster you have to recreate the sequence database with `createdb` and call `cluster` afterwards. You can use update this clustering than. Sorry for the inconvenience. Please open another issue if there is still a problem.

**Comment by martin-steinegger** [comment]
@altaetran does the problem also occur with the newest version MMseqs2? 
Is the issue related to updating?

**Comment by altaetran** [comment]
I'm not sure, I installed a few months ago at most though, and I don't have the flexibility to reinstall a new mmseqs version at the moment.

**Comment by altaetran** [comment]
I manually removed the redundancies before entering into mmseqs and it worked again. I suspect there is something off about the redundancy filter in mmseqs.
